### ISSUE TITLE
Accessibility: Upload Image does not have focus state

### ISSUE THREAD

**Issue by alvaromontoro** [forem/forem#15234]
**Describe the bug**

The "upload image" button does not have a focus state (e.g. an outline on focus), so keyboard users may be confused as the visible focus indicator is lost.

**To Reproduce**

Instructions using keyboard (but can be done with mouse until step 3):

1. Go to https://dev.to
2. Tab to "Create Post" button and press enter
3. Tab several times. The order of the focused elements is as follows:
     1. DEV logo
     2. Edit button
     3. Preview button
     4. Close button
     5. Upload image button
     6. Text editor

You will notice that the "Upload image" element is the only one that doesn't have an indication that it is the currently focused element. 

**Expected behavior**

The upload image should have an outline on focus, the same way that the other elements do.

**Screenshots**

Animated GIF with a screen recording of the tabbing order:

![dev-bug](https://user-images.githubusercontent.com/2205212/139584748-2bb8c21c-bf34-4a88-aee3-b63d565477ae.gif)

Between the Close button and the editor in itself, the focus is lost, because it should be on the Upload Image button but it is not visible.

**Desktop (please complete the following information):**

- OS, version: MacOS 11.6
- Browser, version: Chrome 95

**Additional context**

Screen readers will announce the button, but keyboard-only users (no screen reader) will not know when the focus went.

This issue happens because it's not a button but a label with an input inside. The input in itself is accessibly hidden, what's visible is the label. So when the input gets focused, no outline is displayed (because the input itself is not visible). 

**A possible solution** would be to add styles to the label using `:focus-within`:

```css
label.class-of-the-label:focus-within {
  outline: 2px solid color;
}
```

That way, as the input is inside the label, when the input gets focused, the label will have the outline.
